### ISSUE TITLE
Problem Whatsapp::hasWhatsapp with multiple contacts

### ISSUE THREAD

**Issue by cok28rus** [Auties00/Whatsapp4j#211]

**Comment by Auties00** [comment]
Can you some code/examples?

**Comment by cok28rus** [comment]
Need replace

```
public CompletableFuture<List<HasWhatsappResponse>> hasWhatsapp(@NonNull ContactJidProvider @NonNull ... chats) {
    var contactNodes = Arrays.stream(chats)
        .map(jid -> Node.of("contact", jid.toJid().toPhoneNumber()))
        .toArray(Node[]::new);

    return socketHandler.sendInteractiveQuery(Node.of("contact"),
            Node.ofChildren("user", contactNodes))
        .thenApplyAsync(nodes -> nodes.stream()
            .map(HasWhatsappResponse::new)
            .toList());
  }
```

for

```
public CompletableFuture<List<HasWhatsappResponse>> hasWhatsapp(@NonNull ContactJidProvider @NonNull ... chats) {
    var contactNodes = Arrays.stream(chats)
        .map(jid -> Node.ofChildren("user", Node.of("contact", jid.toJid().toPhoneNumber())))
        .toArray(Node[]::new);

    return socketHandler.sendInteractiveQuery(Node.of("contact"), contactNodes)
        .thenApplyAsync(nodes -> nodes.stream()
            .map(HasWhatsappResponse::new)
            .toList());
  }
```

**Comment by Auties00** [comment]
> hasWhatsapp

That's not right though. I just tested it out

**Comment by cok28rus** [comment]
> > hasWhatsapp
> 
> That's not right though. I just tested it out

Have you tested my version?

**Comment by Auties00** [comment]
> > > hasWhatsapp
> > 
> > 
> > That's not right though. I just tested it out
> 
> Have you tested my version?

Yep, it will always throw a TimeoutException as Whatsapp doesn't know how to handle it

**Comment by cok28rus** [comment]
```
public CompletableFuture<List<HasWhatsappResponse>> hasWhatsapp(@NonNull ContactJidProvider @NonNull ... chats) {
    var contactNodes = Arrays.stream(chats)
        .map(jid -> Node.ofChildren("user", Node.of("contact", jid.toJid().toPhoneNumber())))
        .toArray(Node[]::new);

    return socketHandler.sendInteractiveQuery(Node.of("contact"), contactNodes)
        .thenApplyAsync(nodes -> nodes.stream()
            .map(HasWhatsappResponse::new)
            .toList());
  }
```

This code works, I tested

**Comment by Auties00** [comment]
> 

I'm telling you it doesn't

> public CompletableFuture<List<HasWhatsappResponse>> hasWhatsapp(@NonNull ContactJidProvider @NonNull ... chats) {
>     var contactNodes = Arrays.stream(chats)
>         .map(jid -> Node.ofChildren("user", Node.of("contact", jid.toJid().toPhoneNumber())))
>         .toArray(Node[]::new);
> 
>     return socketHandler.sendInteractiveQuery(Node.of("contact"), contactNodes)
>         .thenApplyAsync(nodes -> nodes.stream()
>             .map(HasWhatsappResponse::new)
>             .toList());
>   }

Well you are right it does work, but so does the other(at least for one contact). Are you querying multiple?

**Comment by Auties00** [comment]
Pushed it

**Comment by Auties00** [comment]
Released 3.1
